### PR TITLE
Added Email Verification

### PR DESCRIPTION
**Pull request by pulkitchauhan03**
In settings.py
- Added settings required to send the Email
- The credentials of the gmail account used to send the mail is set in env.bat for windows or .env for mac (Added to .gitignore)

In register view
- Encoded the user id to send and retrieve over network and to easily query user
- Generated a token out of user id and timestamp in utils.py using PasswordResetTokenGenerator
- Sent an Email with the link

In verification view
- Verified the link
- Set user as active

Created utils.py
- Added a token generator

Changed requirements.txt
- Changed the version of asgiref to asgiref==3.4.1 (As the previous one was not working for me)

### CODE REVIEW

File: register/views.py [hunk 1]
```diff
@@ -16,8 +25,32 @@ def register(request):
...
+
+            email = EmailMessage(
+                email_subject,
+                email_body,
+                'testexpenses724@gmail.com',
+                [user.email],
+                )
+
+            email.send(fail_silently=False)
+            messages.success(request, 'Account successfully created')
```
Comment (ExpressHermes) [inline register/views.py]: The message should be 'Account successfully created. Please check your email for the verification link'

**Review by ExpressHermes — COMMENTED** [review]
Great work. Just do the requested change and I will merge this.

BTW, great description. Clear and Detailed.

**Comment by pulkitchauhan03** [comment]
I changed the wrong message, changed it in the subsequent commit.

**Comment by ExpressHermes** [comment]
@pulkitchauhan03 Have you tested this setup locally. Does it work properly? (Email is being sent, user can log in after activating their account?)

**Comment by pulkitchauhan03** [comment]
I forgot to add six in the requirements.txt . Now it should work. It is working fine for me

**Review by ExpressHermes — APPROVED** [review]

**Pull request merged**